### PR TITLE
Fix infrastructure tests after gardener v1.56.0

### PR DESCRIPTION
**Pull request by kon-angelo**
**How to categorize this PR?**

/area control-plane
/kind bug
/platform aws

**What this PR does / why we need it**:

After revendoring of gardener `v1.56.0` the priority class is necessary for the terraformer pod.

/invite @shafeeqes 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator

```

### CODE REVIEW

File: test/integration/infrastructure/infrastructure_test.go [hunk 1]
```diff
@@ -313,6 +316,16 @@ var _ = Describe("Infrastructure tests", func() {
...
+			priorityClass = &schedulingv1.PriorityClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: v1beta1constants.PriorityClassNameShootControlPlane300,
+				},
+				Description:   "PriorityClass for Shoot control plane components",
+				GlobalDefault: false,
+				Value:         999998300,
+			}
+			Expect(client.IgnoreAlreadyExists(c.Create(ctx, priorityClass))).To(BeNil())
+
```
Comment (shafeeqes) [inline test/integration/infrastructure/infrastructure_test.go:328]: Shouldn't this be inside the `runTest` function here,
https://github.com/gardener/gardener-extension-provider-aws/blob/5e67cfc2354fdd1017479452589893894f7b4201/test/integration/infrastructure/infrastructure_test.go#L367
I see lot of code is duplicated here which is in runTest, can we reuse them?

Comment (dkistner) [reply test/integration/infrastructure/infrastructure_test.go:328]: Can we not simply put this logic into `BeforeSuite`? It's a global/non namespaced resource, which is needed by multiple test cases.

Comment (shafeeqes) [reply test/integration/infrastructure/infrastructure_test.go:328]: Yes. Better.

**Review by dkistner — APPROVED** [review]

**Pull request merged**